http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -294,6 +328,11 @@ public void abort() {
          * when constructing the SSL context.
          */
         Builder tlsTrustManagersProvider(TlsTrustManagersProvider tlsTrustManagersProvider);
+
+        /**
+         * Configuration that defines how to communicate via an HTTP proxy.
+         */
+        Builder proxyConfiguration(ProxyConfiguration proxyConfiguration);
```

Comment:
Can we add [Builder with consumer](https://github.com/aws/aws-sdk-java-v2/blob/ab747a55116c45788c16390d8c2d280bb063bc01/http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java#L304) also ?